plugins/woocommerce/src/Admin/WCAdminHelper.php

```diff
@@ -122,4 +122,115 @@ public static function is_site_fresh() {
 
 		return $date > $month_ago;
 	}
+
+	/**
+	 * Test if a URL is a store page. This function ignores the domain and protocol of the URL and only checks the path and query string.
+	 *
+	 * Store pages are defined as:
+	 *
+	 * - My Account
+	 * - Shop
+	 * - Cart
+	 * - Checkout
+	 * - Privacy Policy
+	 * - Terms and Conditions
+	 *
+	 * Additionally, the following autogenerated pages should be included:
+	 * - Product pages
+	 * - Product Category pages
+	 * - Product Tag pages
+	 *
+	 * @param string $url URL to check. If not provided, the current URL will be used.
+	 * @return bool Whether or not the URL is a store page.
+	 */
+	public static function is_store_page( $url = '' ) {
+		$url = $url ? $url : esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ?? '' ) );
+
+		if ( ! $url ) {
+			return false;
+		}
+		$normalized_path = self::get_normalized_url_path( $url );
+
+		// WC store pages.
+		$store_pages = array(
+			'myaccount' => wc_get_page_id( 'myaccount' ),
+			'shop'      => wc_get_page_id( 'shop' ),
+			'cart'      => wc_get_page_id( 'cart' ),
+			'checkout'  => wc_get_page_id( 'checkout' ),
+			'privacy'   => wc_privacy_policy_page_id(),
```

Comment:
not sure how big of an impact it is on performance but there's a get_option() call in wc_privacy_policy_page_id()